flow/workflows/qrep_flow.go

```diff
@@ -365,6 +394,30 @@ func QRepFlowWorkflow(
 		"Last Partition", lastPartition,
 		"Number of Partitions Processed", numPartitionsProcessed)
 
+	// here, we handle signals after the end of the flow because a new workflow does not inherit the signals
+	// and the chance of missing a signal is much higher if the check is before the time consuming parts run
+	q.receiveAndHandleSignal(ctx)
+	if q.activeSignal == shared.ShutdownSignal {
```

Comment:
Can move this comparison after the PauseSignal comparison & then remove exit handling from being duplicated after PauseSignal for loop